app/sagas/selectServer.js

```diff
@@ -36,8 +37,21 @@ const handleSelectServer = function* handleSelectServer({ server, version, fetch
 		if (fetchVersion) {
 			serverInfo = yield getServerInfo({ server });
 		}
-		yield AsyncStorage.setItem('currentServer', server);
-		const userStringified = yield AsyncStorage.getItem(`${ RocketChat.TOKEN_KEY }-${ server }`);
+
+		const { serversDB } = database.databases;
+		const servers = yield serversDB.objects('servers');
+		serversDB.write(() => {
+			try {
+				// eslint-disable-next-line no-plusplus
+				for (let i = 0; i < servers.length; i++) {
+					serversDB.create('servers', { id: servers[i].id, currentServer: servers[i].id === server }, true);
+				}
+			} catch (e) {
+				log('err_update_current_server ->', e);
+			}
+		});
+
+		const { username: userStringified } = yield Keychain.getInternetCredentials(server);
```

Comment:
for safety reasons :) `const { username: userStringified = null } = server && yield Keychain.getInternetCredentials(server);`